### PR TITLE
Feature/eicnet 799 author of a discussion is also shown in the list of contributors

### PR DESCRIPTION
**Pull request by fonteijne**

### CODE REVIEW

File: lib/modules/eic_content/modules/eic_content_discussion/eic_content_discussion.module [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function eic_content_discussion_node_insert(\Drupal\node\NodeInterface $entity) {
+  if ('discussion' === $entity->bundle()) {
+    \Drupal::classResolver(EntityOperations::class)
+      ->discussionInsert($entity);
+  }
+}
```
Comment (4hardfire) [inline lib/modules/eic_content/modules/eic_content_discussion/eic_content_discussion.module:22]: Hi @fonteijne,
This works like a charm, however I would suggest implementing the logic using the **hook_entity_presave()**. This way you avoid saving the entity twice.

**Review by 4hardfire — CHANGES_REQUESTED** [review]

File: lib/modules/eic_content/modules/eic_content_discussion/eic_content_discussion.module (hunk 1 above)
Comment (fonteijne) [reply lib/modules/eic_content/modules/eic_content_discussion/eic_content_discussion.module:22]: Nice! Just a note from our call:
use the `isNew()` to determine whether the entity is created or updated.

Comment (fonteijne) [reply lib/modules/eic_content/modules/eic_content_discussion/eic_content_discussion.module:22]: - [ ] Use `hook_entity_presave()`
- [ ] Use `isNew()` to on run on the "insert" operation.

**Review by 4hardfire — APPROVED** [review]
This seems ok now.

**Pull request merged**